Miracast/MiracastPlayer/MiracastGstPlayer.cpp

```diff
@@ -0,0 +1,932 @@
+/*
+ * If not stated otherwise in this file or this component's Licenses.txt file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gst/audio/audio.h>
+#include <gst/app/gstappsink.h>
+#include <unistd.h>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <sys/types.h>
+#include <sys/syscall.h>
+#include "MiracastLogger.h"
+#include "MiracastController.h"
+#include "MiracastGstPlayer.h"
+
+MiracastGstPlayer *MiracastGstPlayer::mMiracastGstPlayer{nullptr};
+
+MiracastGstPlayer *MiracastGstPlayer::getInstance()
+{
+    if (mMiracastGstPlayer == nullptr)
+    {
+        mMiracastGstPlayer = new MiracastGstPlayer();
+    }
+    return mMiracastGstPlayer;
+}
+
+void MiracastGstPlayer::destroyInstance()
+{
+    MIRACASTLOG_TRACE("Entering...");
+    if (mMiracastGstPlayer != nullptr)
+    {
+        mMiracastGstPlayer->stop();
+        if (mMiracastGstPlayer->stop())
+        {
+            MIRACASTLOG_ERROR("Stop miracast player");
```

Comment:
change to info

Reply:
Addressed in https://github.com/rdkcentral/rdkservices/pull/4618/commits/abb5db62d2585d5e30a40d8f7c1645336732c0ea